craft_application/commands/lifecycle.py

```diff
@@ -370,3 +401,23 @@ def _launch_shell() -> None:
     emit.progress("Launching shell on build environment...", permanent=True)
     with emit.pause():
         subprocess.run(["bash"], check=False)
+
+
+def _run_managed(destructive_mode: str | None) -> bool:
+    """Return whether the command should run in managed mode or not.
+
+    The command will run in managed mode unless the `--destructive-mode` flag
+    is passed OR `CRAFT_BUILD_ENVIRONMENT` is set to `host`.
+    """
+    if destructive_mode:
+        emit.debug("Not running managed mode because `--destructive-mode` was passed")
+        return False
+
+    build_env = os.getenv("CRAFT_BUILD_ENVIRONMENT")
```

Comment:
Is it intentional that `CRAFT_BUILD_ENVIRONMENT=host` is different from `--destructive-mode`?

Reply:
different in what sense?

Reply:
Hmm actually I was wrong. We used to have some things referencing destructive mode in `application.py`, but it looks like this has been cleaned up so `command.run_managed()` is the only thing that cares about it. That's a very pleasant surprise for me! 🎉